course/models.py

```diff
@@ -477,10 +477,10 @@ def is_valid_language(self, lang):
     @property
     def default_language(self):
         language = self.language
-        language_list = [x for x in language.split("|") if x]
+        language_list = language.strip('|').split('|')
```

Comment:
lstrip and strip are different, and `split(str, number)` limits how many times the split is executed, we need to find only the first.

Reply:
Interestingly. the `split('|')` is faster with 2 languages, but slower with more (where `split('|', 1)` is faster :thinking:

Reply:
Now this code does not work with an empty `self.language`:
`"".strip("|").split("|")  -> ['']`
It is a list with one element, the empty string, so the list evaluates to True in the if condition and an empty language would be returned.

Reply:
It's missing the `[0]`
```python3
>>> ''.lstrip('|').split('|', 1)[0]
''
>>> 'fi'.lstrip('|').split('|', 1)[0]
'fi'
>>> '|en|fi|'.lstrip('|').split('|', 1)[0]
'en'
```